R/check_tbl_spl_n.R

```diff
@@ -14,21 +15,56 @@
 #' See [hubverse documentation on samples](https://hubverse.io/en/latest/user-guide/sample-output-type.html)
 #' for more details.
 #' @export
-check_tbl_spl_n <- function(tbl, round_id, file_path, hub_path) {
+check_tbl_spl_n <- function(tbl, round_id, file_path, hub_path,
+                            compound_taskid_set = NULL) {
+  if (!is.null(compound_taskid_set) && isTRUE(is.na(compound_taskid_set))) {
+    cli::cli_abort("Valid {.var compound_taskid_set} must be provided.")
+  }
   config_tasks <- hubUtils::read_config(hub_path, "tasks")
+  if (is.null(compound_taskid_set)) {
+    compound_taskid_set <- get_round_compound_task_ids(
+      config_tasks,
+      round_id
+    )
+  }
 
   if (isFALSE(has_spls_tbl(tbl)) || isFALSE(hubUtils::is_v3_config(config_tasks))) {
     return(skip_v3_spl_check(file_path))
   }
 
-  hash_tbl <- spl_hash_tbl(tbl, round_id, config_tasks)
+  hash_tbl <- spl_hash_tbl(tbl, round_id, config_tasks, compound_taskid_set)
   n_ranges <- get_round_spl_n_ranges(config_tasks, round_id)
 
   n_tbl <- dplyr::group_by(hash_tbl, .data$compound_idx) %>%
     dplyr::summarise(
       n = dplyr::n_distinct(.data$output_type_id),
       mt_id = unique(.data$mt_id)
-    ) %>%
+    )
+
```

Comment:
The following splits the previous check that only looked at whether samples for each compound idx fall within the expected range into:
1. check that all  compound idxs have the same number of samples
2. If above check passes, check that the number of samples falls within the expected range